src/auth/esritoken/CMakeLists.txt

```diff
@@ -1,38 +1,59 @@
 set(AUTH_ESRI_TOKEN_SRCS
   qgsauthesritokenmethod.cpp
-  qgsauthesritokenedit.cpp
 )
+if (WITH_GUI)
+  set(AUTH_ESRI_TOKEN_SRCS ${AUTH_ESRI_TOKEN_SRCS}
+    qgsauthesritokenmethodgui.cpp
+    qgsauthesritokenedit.cpp
+  )
+endif()
 
 set(AUTH_ESRI_TOKEN_HDRS
-  qgsauthesritokenedit.h
   qgsauthesritokenmethod.h
 )
-
-set(AUTH_ESRI_TOKEN_UIS qgsauthesritokenedit.ui)
+if (WITH_GUI)
+  set(AUTH_ESRI_TOKEN_HDRS ${AUTH_ESRI_TOKEN_HDRS}
+    qgsauthesritokenedit.h
+  )
+endif()
 
 include_directories (
   ../../core
   ../../core/auth
   ../../core/geometry
   ../../core/metadata
```

Comment:
I believe that you can remove these includes now that we use target_link_librairies to qgis_core

Reply:
@3nids , good one.